packages/core/src/invoke.ts

```diff
@@ -151,20 +156,47 @@ export function spawnPromise<T>(
   };
 }
 
+export function spawnActivity<TC, TE extends EventObject>(
+  activityCreator: (ctx: TC, event: TE) => any
+): InvokeCreator<TC, TE> {
+  return (ctx, e, { parent, id }) => {
+    let dispose;
+    try {
+      dispose = activityCreator(ctx, e);
+    } catch (err) {
+      parent.send(error(id, err) as any);
+    }
+
+    return {
+      id,
+      send: () => void 0,
+      toJSON: () => ({ id }),
+      subscribe() {
+        // do nothing
+        return {
+          unsubscribe: () => void 0
+        };
+      },
+      stop: isFunction(dispose) ? () => dispose() : undefined
+    };
+  };
+}
+
 export function spawnCallback<TE extends EventObject = AnyEventObject>(
-  callback: (ctx: any, e: any) => InvokeCallback
+  callbackCreator: (ctx: any, e: any) => InvokeCallback
 ): InvokeCreator<any, AnyEventObject> {
-  return (_, __, { parent, id }) => {
+  return (ctx, event, { parent, id, _event }) => {
+    const callback = callbackCreator(ctx, event);
     let canceled = false;
     const receivers = new Set<(e: EventObject) => void>();
     const listeners = new Set<(e: EventObject) => void>();
 
-    const receive = (e: TE) => {
-      listeners.forEach(listener => listener(e));
+    const receive = (receivedEvent: TE) => {
+      listeners.forEach(listener => listener(receivedEvent));
       if (canceled) {
```

Comment:
this is an important semantic to establish - is it OK for a stopped actor to still emit events to the outside world? shouldnt the `canceled` check be first here to prevent notifying listeners in such case? should current listeners be notified about an actor is being stopped?

Reply:
That's up to the implementor. In the real world, there's nothing stopping actors from emitting events. The best the receiver can do is ignore those events.

Reply:
well - if we plan to remove `subscribe` from actors as discussed on Slack then this comment is irrelevant, actor->parent communication is already guarded in a way I would expect it to

Reply:
Right, it will be